src/space-between/internal.tsx

```diff
@@ -25,22 +29,26 @@ export default function InternalSpaceBetween({
   const flattenedChildren = flattenChildren(children);
 
   return (
-    <div
+    <Variant
       {...baseProps}
       className={clsx(baseProps.className, styles.root, styles[direction], styles[`${direction}-${size}`])}
       ref={__internalRootRef}
     >
       {flattenedChildren.map(child => {
-        // If this react child is a primitive value, the key will be undefined
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        const key = (child as any).key;
+        const key = typeof child === 'object' ? child.key : undefined;
+
+        // When using "ul" variant we avoid extra element wrappers to keep list semantics.
+        if (Variant === 'ul' && typeof child === 'object') {
+          const className = clsx(child.props.className, styles.child, styles[`child-${direction}-${size}`]);
+          return React.cloneElement(child, { className });
```

Comment:
I kind of like the way it works - instead of creating additional div elements this applies styles directly to passed children. However, it requires complex assumptions: children must propagate className; children styles must not define margins.

I wonder if there is a better solution for this. E.g. we can create an internal List element and use it interchangeable. However, that would require copying space-between styles. Wdyt?

Reply:
In case we create a separate list component it would require a slightly different API from space-between if we want to avoid complex assumptions on React children. In that case using space-between and list-space-between interchangeably could be difficult. However, I am not sure if we need to - how about making the token group a list by default?

Reply:
I agree TokenGroup could be a list by default